### ISSUE TITLE
counsel-ag matcher always fuzzy

### ISSUE THREAD

**Issue by diamond-lizard** [abo-abo/swiper#1214]
When I start counsel-ag the default matcher is `fuzzy`, but when I try to change it to `ivy` or `order` by typing `C-o m` then the text displayed under the action header in the hydra changes to `ivy` or `order` as expected, but the actual matching itself stays fuzzy.

The same matcher changes work just fine when I hit `C-o m` in other contexts, such as in the directory chooser under the `ag` command itself.

I don't know if this makes any difference, but I do have [flx](https://github.com/lewang/flx.git) installed, and have this in my emacs init:

```
(setq ivy-re-builders-alist
      '((t . ivy--regex-fuzzy)))
```

**Comment by basil-conto** [comment]
> but the actual matching itself stays fuzzy

WDYM by "fuzzy" here? Can you give step-by-step example of what you'd expect vs what actually happens?

IIUC you'd have to translate Emacs regexps to a pattern format understood by `ag` in order for Ivy's concept of regexp builders to carry over, which I'm not sure would even be possible in the general case. In other words, my understanding is that input to `counsel-ag` should be in a format understood by `ag` and independent of your `ivy-re-builders-alist`.

**Comment by diamond-lizard** [comment]
Unfortunately, it's been too long since I filed this issue, and I can't remember exactly what I meant by that.  If I had to guess, I probably meant that the `fuzzy` matching I was referring to matched as if there was a `.*` between every character, and that behavior persisted even when the other matching types were used.  But I can't reproduce that behavior now, and I'm not sure which version of counsel-ag I was using when I filed that bug report in the first place.

I think this issue can be closed, and if I'll just open a new issue if I run in to this behavior again, and next time I'll give more detail and step-by-step instructions for reproducing the issue, along with a description of what I expected to happen vs what actually happened.
